testsuite/tests/lib-channels/input_all.ml

```diff
@@ -0,0 +1,72 @@
+(* TEST
+readonly_files = "input_all.ml"
+*)
+
+let data_file =
+  "data.txt"
+
+let random_string size =
+  String.init size (fun _ -> Char.chr (Random.int 256))
+
+(* various sizes, binary mode *)
+
+let check size =
+  let data = random_string size in
+  Out_channel.with_open_bin data_file (fun oc -> Out_channel.output_string oc data);
+  let read_data = In_channel.with_open_bin data_file In_channel.input_all in
+  assert (data = read_data)
+
+let () =
+  List.iter check [ 0; 1; 65536; 65536 + 1; 2 * 65536 ]
+
+(* binary mode; non-zero starting position *)
+
+let data_size = 65536
+
+let check midpoint =
+  let data = random_string data_size in
+  Out_channel.with_open_bin data_file
+    (fun oc -> Out_channel.output_string oc data);
+  let contents =
+    In_channel.with_open_bin data_file
+      (fun ic ->
+         let s1 = Option.get (In_channel.really_input_string ic midpoint) in
+         let s2 = In_channel.input_all ic in
+         s1 ^ s2
+      )
+  in
+  assert (contents = data)
+
+let () =
+  List.iter check [0; 1; 100; data_size]
+
+(* text mode *)
+
+(* translates LF into CRLF in-place in [fn] *)
+let unix2dos fn =
+  let s = In_channel.with_open_text fn In_channel.input_all in
+  Out_channel.with_open_text fn
+    (fun oc -> Out_channel.output_string oc s)
+
+let source_fn =
+  "input_all.ml"
```

Comment:
On Windows, whether `input_all.ml` has unix or dos end of lines will depend on how the user configures git. Does that matter here?

Reply:
Indeed, if you make a CRLF checkout, then that would make the test fail. I pushed a change to make the test work also in this case.